src/core/schema.ts

```diff
@@ -29,6 +30,22 @@ export const defaultSchema: Schema = {
     // [0-9]
     ...objectFromEntries("0123456789".split("").map((n) => [n, n])),
   },
+  defaultEventNames: {
+    a: () => "click",
+    button: () => "click",
+    form: () => "submit",
+    details: () => "toggle",
+    input: (e) => (e.getAttribute("type") == "submit" ? "click" : "input"),
+    select: () => "change",
+    textarea: () => "input",
+  }
+}
+
+export function getDefaultEventNameForElement(element: Element, schema = defaultSchema): string | undefined {
```

Comment:
Maybe this event should be an instance or static method on the Application instead?

While we are moving things around. Gives even more control if needed on custom Applications.

For example, if we wanted to make the default event different for a specific controlled element we could access the controller identifier in the function by `elemet.hasAttribute(this.schema.controllerIdentifier,...);`

Reply:
That would also require the `Action` constructor to receive an Application instance where as it currently receives only the schema.

---

However, I do think it would be nice to make this a method on schema, so I've made that change.

So now instead of this:

```ts
import {defaultSchema, Application} from '@hotwired/stimulus'

const schema = {
  ...defaultSchema,
  defaultEventNames: {
    ...defaultSchema.defaultEventNames,
    'sl-button': () => 'click',
    'sl-select': () => 'change',
    'sl-input': () => 'input',
  }
}

const application = new Application.start(document.documentElement, schema)
```

Consumers would do this:

```ts
import {defaultSchema, Application} from '@hotwired/stimulus'

const customElementDefaultEventNames = {
  'sl-button': 'click',
  'sl-select': 'change',
  'sl-input': 'input',
}

const schema = {
  ...defaultSchema,
  getDefaultEventNameForElement: (element) => (
    defaultSchema.getDefaultEventNameForElement(element) ||
    customElementDefaultEventNames[element.tagName.toLowerCase()]
  )
}

const application = new Application.start(document.documentElement, schema)
```

Reply:
I tried this, but it wasn't really that nice (see https://github.com/hotwired/stimulus/pull/661#issuecomment-1592881090).